ADCIO-2215) refactor: migration gradle convention plugin

aahspringaa4 opened the pull request:
♻️ Current situation
*  기존 gradle의 역할
   * module dependencies 관리
   * App Compile & SDK 버전 관리
   * mavencentral 배포 정보 관리

🎲 Problem
1. gradle에서 처리하는 역할이 많았기에 의존도가 높았음 (플러그인 모듈이라 특히)
2. 사용하는 sdk나 라이브러리 버전 변경 시 일일이 반영해주어야했음
3. 새로운 개발자에게 인수인계에 어려움
4. 빌드 시간 증가, gradle sync 시간 증가

💡 Proposed solution
* Convention Plugin 분리로 gradle 마다 공통 되는 모듈을 convention 처럼 한번에 정의 
   * 효율성 극대화
   * 빌드 시간 대폭 감소 (추후 정확한 수치로 안내드릴꼐요!)
 * ktlint & detekt 설정
   * 빌드하면 자동적으로 lint 통일해 줌
  띄어쓰기, trailling comma, 줄 바꿈 등
  
 🎹 CC. 커밋이 많은 이유
-> 로컬 환경에는 없는 CI에서만 발생하는 문제 테스트를 위함 (files changed로 봐주세요!)

aahspringaa4 committed:
build: implement build-logic gradle convention plugin module gradle

aahspringaa4 committed:
build: add local enum module gradle

aahspringaa4 committed:
feat: add convention enum object

aahspringaa4 committed:
feat: add plugin enum object

aahspringaa4 committed:
feat: add application constants

aahspringaa4 committed:
build: add build logic gradle's implement

aahspringaa4 committed:
feat: add gradle plugin setting for local maven

aahspringaa4 committed:
feat: add application setting convention

aahspringaa4 committed:
style: add code enter

aahspringaa4 committed:
feat: add compose convention file

aahspringaa4 committed:
feat: add convention extension module provider

aahspringaa4 committed:
build: refactor root setting gradle

aahspringaa4 committed:
feat: add application version setting convention plugin

aahspringaa4 committed:
build: add libs version toml file

aahspringaa4 committed:
feat: application version name init

aahspringaa4 committed:
docs: add local enum publishing note

aahspringaa4 committed:
build: remove ktlint check in ci

aahspringaa4 committed:
build: update agent gradle by convention plugin

aahspringaa4 committed:
build: update plugin module gradle by convention plugin

aahspringaa4 committed:
feat: add local-enums to local maven

aahspringaa4 committed:
build: update toml version files

aahspringaa4 committed:
build: update common project gradle setting

set ktlint & detekt

aahspringaa4 committed:
build: add jvm provide convention plugin

aahspringaa4 committed:
build: add viewbinding convention plugin

aahspringaa4 committed:
build: add compose convention plugin

aahspringaa4 committed:
build: add library module's convention

aahspringaa4 committed:
feat: add view configure in application convention

aahspringaa4 committed:
feat: add viewbinding plugin convention

aahspringaa4 committed:
feat: add compose plugin convention

aahspringaa4 committed:
build: update application version

aahspringaa4 committed:
feat: add viewbinding convention key

aahspringaa4 committed:
build: change prefix key value

aahspringaa4 committed:
feat: register convention plugin by enum key

aahspringaa4 committed:
build: update app module gradle by gradle convention

limsaehyun reviewed (commented):
ooh.. this is cool :-)

aahspringaa4 committed:
build: include local-enums module in build

aahspringaa4 committed:
build: remove don't use dokka library

aahspringaa4 committed:
build: remove ktlint check in build because not unusage

aahspringaa4 committed:
build: update core library dependencies

aahspringaa4 committed:
build: update placement gradle dependencies to convention plugin

aahspringaa4 committed:
build: change file package language

aahspringaa4 committed:
build: update app constant version

aahspringaa4 committed:
build: delete suspected errors

aahspringaa4 committed:
Revert "build: change file package language"

This reverts commit e849d058fa995c999bba717b3c2c6400b412ea69.

aahspringaa4 committed:
refactor: get outside local enums module

aahspringaa4 committed:
test: remove don't use test

aahspringaa4 committed:
build: remove lint check application gradle convention

aahspringaa4 committed:
build: update sdk version 33 -> 34

aahspringaa4 committed:
build: remove ktlint extension auto checking in project gradle

aahspringaa4 committed:
build: add set up local enums local publish in ci

aahspringaa4 committed:
build: change ci build order

aahspringaa4 committed:
style: check ktlint format

aahspringaa4 committed:
fix: add a build part that obfuscates okhttp to modules that use the okhttp.

aahspringaa4 committed:
feat: add local maven route

ci test commit #1

aahspringaa4 committed:
build: change version setting method

aahspringaa4 committed:
build: remove build-logic group router

ci test commit #3

aahspringaa4 committed:
build: add local library setting

aahspringaa4 committed:
build: change version shape to semi version

aahspringaa4 committed:
build: change ci seqence

aahspringaa4 committed:
build: reset changed things for fix ci

aahspringaa4 committed:
build: set enum class version

aahspringaa4 committed:
build: change build include seqence

aahspringaa4 committed:
build: separte module's build in ci

aahspringaa4 committed:
build: update ci error

aahspringaa4 committed:
build: fix ci error

aahspringaa4 committed:
build: update ci code

aahspringaa4 committed:
build: update ci code

aahspringaa4 committed:
build: add cache clear logic

aahspringaa4 committed:
build: update ci code

aahspringaa4 committed:
build: update ci code

aahspringaa4 committed:
build: update module version

aahspringaa4 committed:
build: update ci

aahspringaa4 committed:
refactor: set local-enums in build-logic

aahspringaa4 committed:
build: update ci

aahspringaa4 committed:
build: update ci

aahspringaa4 committed:
build: remove don't use logic

aahspringaa4 committed:
build: change ci build first

aahspringaa4 committed:
build: update ci code

aahspringaa4 committed:
build: update module compile

aahspringaa4 committed:
build: add group id in gradle

aahspringaa4 committed:
refactor: change file location

aahspringaa4 committed:
build: remove don't use code

aahspringaa4 committed:
build: update plugin version

aahspringaa4 committed:
build: remove ktlint extension auto checking in project gradle

build: add set up local enums local publish in ci

build: change ci build order

style: check ktlint format

fix: add a build part that obfuscates okhttp to modules that use the okhttp.

feat: add local maven route

ci test commit #1

build: change version setting method

build: remove build-logic group router

ci test commit #3

build: add local library setting

build: change version shape to semi version

build: change ci seqence

build: reset changed things for fix ci

build: set enum class version

build: change build include seqence

build: separte module's build in ci

build: update ci error

build: fix ci error

build: update ci code

build: update ci code

build: add cache clear logic

build: update ci code

build: update ci code

build: update module version

build: update ci

refactor: set local-enums in build-logic

build: update ci

build: update ci

build: remove don't use logic

build: change ci build first

build: update ci code

build: update module compile

build: add group id in gradle

refactor: change file location

build: remove don't use code

build: update plugin version

aahspringaa4 committed:
Merge branch 'feature/migration-gradle-convention-plugins' of https://github.com/corca-ai/adcio_android_plugins into feature/migration-gradle-convention-plugins

aahspringaa4 committed:
build: clean up ci

aahspringaa4 committed:
Revert "build: clean up ci"

This reverts commit 5cf0cc27bcfe558c38d0e33739cd2217b90a18fa.

aahspringaa4 committed:
build: remove local enum route

aahspringaa4 committed:
build: upload local maven is first

aahspringaa4 committed:
Revert "build: upload local maven is first"

This reverts commit e4dfb802044f2bafefe686a8c24dd99a8b323f7f.

aahspringaa4 committed:
build: clear ci cache before local maven publish

aahspringaa4 committed:
build: add local maven setting gradle

aahspringaa4 committed:
buiild: update ci

aahspringaa4 committed:
build: remove clear cache logic

aahspringaa4 committed:
build: change local maven senematic version

aahspringaa4 committed:
build: update ci

aahspringaa4 committed:
buiild: remove build cache after 7days

aahspringaa4 committed:
build: update ci

aahspringaa4 committed:
build: set publish setting

aahspringaa4 committed:
build: add gradle clear

aahspringaa4 committed:
build: change dependency directly

aahspringaa4 committed:
build: set build-logic module language version

aahspringaa4 committed:
build: change convention enum depend method

aahspringaa4 committed:
build: change local setting

aahspringaa4 committed:
build: set env setting in ci

aahspringaa4 committed:
build: clean up ci

aahspringaa4 committed:
build: update ci

aahspringaa4 committed:
fix: fix ci error

aahspringaa4 committed:
build: update ci

aahspringaa4 committed:
build: update gradle

aahspringaa4 committed:
build: update ci

aahspringaa4 committed:
build: update gradle

aahspringaa4 committed:
build: update gradle

aahspringaa4 committed:
build: update build file

aahspringaa4 committed:
build: update build file

aahspringaa4 committed:
build: update gradle properties

aahspringaa4 committed:
build: add ci debug

aahspringaa4 committed:
build: update ci

aahspringaa4 committed:
build: update ci

aahspringaa4 committed:
build: update ci

aahspringaa4 committed:
build: add setting gradle set

aahspringaa4 committed:
build: update gradle extension

aahspringaa4 committed:
build: update version

aahspringaa4 committed:
Revert "build: update version"

This reverts commit 78d945bccc8d0998f7f6f82b5773e8c9bb15f374.

aahspringaa4 committed:
build: update ci

aahspringaa4 committed:
build: change ci seqence

aahspringaa4 committed:
build: change ci simple

aahspringaa4 committed:
Revert "build: change ci simple"

This reverts commit d2ba7c99ea437043e685a1a6739bfc9a1a07785c.

aahspringaa4 committed:
build: change ci

aahspringaa4 committed:
build: update ci

aahspringaa4 committed:
build: update ci

aahspringaa4 committed:
build: update ci

aahspringaa4 committed:
build: clean by kt lint

aahspringaa4 committed:
build: ci error run

aahspringaa4 committed:
build: add jupiter library in placement
